File: src/EpiserverReact/Static/js/features/content/BannerBlock.js
import HeadingBlock, { HeadingBlockProperty } from '../../foundation/blocks/HeadingBlock';
import LinkBlock, { LinkBlockProperty } from '../../foundation/blocks/LinkBlock';
import ImageMedia, { ImageMediaWrapper } from '../../foundation/media/ImageMedia';
import ContentModel from '../../foundation/models/ContentModel';

export default class BannerBlock extends React.Component {
    render() {

        /** @type {BannerBlockModel} */
        const model = this.props.data;

        return (
            <div>
                <div>
                    <ImageMedia data={model.background.expandedValue} />
                </div>
                <div>
                    <HeadingBlock data={model.heading.value} />
                </div>
                <div>
                    <LinkBlock data={model.callToAction.value} />
                </div>
            </div>
        );
    }
}

export class BannerBlockModel extends ContentModel {

    /** @type {ImageMediaWrapper} */
    background;

    /** @type {LinkBlockProperty} */
    callToAction;

    /** @type {HeadingBlockProperty} */
    heading;
}

File: src/EpiserverReact/CMS/Features/Content/Models/BannerBlock.cs
namespace CMS.Features.Content.Models
{
    using System.ComponentModel.DataAnnotations;
    using EPiServer.Core;
    using EPiServer.DataAnnotations;
    using EPiServer.Web;
    using Foundation.Blocks;
    using Foundation.Media;

    [ContentType(DisplayName = "BannerBlock", GUID = "d1c4ceeb-26fa-49e6-bb87-cb0d7e7725e8", Description = "")]
    public class BannerBlock : BlockData
    {
        public virtual HeadingBlock Heading { get; set; }

        public virtual LinkBlock CallToAction { get; set; }

        [AllowedTypes(typeof(ImageMedia))]
        [UIHint(UIHint.Image)]
        public virtual ContentReference Background { get; set; }
    }
}